test/unit/redirect.test.js

```diff
@@ -0,0 +1,211 @@
+/**
+ * Copyright 2023 IBM Corp. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const nock = require('nock');
+const { AxiosError } = require('axios');
+const { NoAuthAuthenticator, BaseService } = require('../../dist');
+
+// Disable real network connections.
+nock.disableNetConnect();
+
+const safeHeaders = {
+  Authorization: 'foo',
+  Cookie: 'baz',
+  Cookie2: 'baz2',
+  'WWW-Authenticate': 'bar',
+};
+
+function initService(url) {
+  const service = new BaseService({
+    authenticator: new NoAuthAuthenticator(),
+    serviceUrl: url,
+  });
+
+  const parameters = {
+    options: {
+      method: 'GET',
+      url: '/',
+      headers: safeHeaders,
+    },
+    defaultOptions: {
+      serviceUrl: url,
+    },
+  };
+
+  return { service, parameters };
+}
+
+describe('Node Core redirects', () => {
+  afterEach(() => {
+    nock.cleanAll();
+  });
+
+  it('should include safe headers within cloud.ibm.com domain', async () => {
+    const url1 = 'http://region1.cloud.ibm.com';
```

Comment:
let's use https in these URLs :)

Reply:
Agreed! I also change the protocols in my Python core PR, because I used HTTP there too.